### PR TITLE
blender: wrap in separate derivation, fixes #97250

### LINKED ISSUE #97250: Blender missing "requests" python library for add-ons
## Issue description

Blender is missing the Python `requests` library which is a dependency of at least one popular add-on

### Steps to reproduce

Open `Edit > Preferences > Add-ons` and select `3D View: BlenderKit Online Asset Library` and this Python backtrace is presented:

```
Traceback (most recent call last):
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/modules/addon_utils.py", line 351, in enable
    mod = __import__(module_name)
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/__init__.py", line 51, in <module>
    from blenderkit import asset_inspector, search, download, upload, ratings, autothumb, ui, icons, bg_blender, paths, \
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/search.py", line 33, in <module>
    from blenderkit import paths, utils, categories, ui, colors, bkit_oauth, version_checker, tasks_queue, rerequests
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/categories.py", line 27, in <module>
    from blenderkit import paths, utils, tasks_queue, rerequests
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/rerequests.py", line 28, in <module>
    from blenderkit import ui, utils, paths, tasks_queue, bkit_oauth
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/ui.py", line 34, in <module>
    from blenderkit import paths, ratings, utils, search, upload, ui_bgl, download, bg_blender, colors, tasks_queue
  File "/nix/store/v8ykwxys87dfh60s414ds70ckbgyl9yx-blender-2.83.1/share/blender/2.83/scripts/addons/blenderkit/ratings.py", line 30, in <module>
    import requests, threading
ModuleNotFoundError: No module named 'requests'
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.7.7, NixOS, 20.03.git.a41b1a288c7 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - nixpkgs: `/home/luke/git/nixpkgs`

### PR DESCRIPTION
**Pull request by FRidh**
- to make it possible to add additional python packages for plugins
- set other wrapper arguments

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
Downside of using a wrapper derivation is that you can no longer easily override the unwrapped derivation. Unless a separate top-level attribute is used, like is done with some other packages.

File: pkgs/applications/misc/blender/default.nix [hunk 1]
```diff
@@ -8,144 +8,168 @@
...
+    };
 
-  # Set RUNPATH so that libcuda and libnvrtc in /run/opengl-driver(-32)/lib can be
-  # found. See the explanation in libglvnd.
-  postFixup = optionalString cudaSupport ''
-    for program in $out/bin/blender $out/bin/.blender-wrapped; do
-      isELF "$program" || continue
-      addOpenGLRunpath "$program"
-    done
+  } # --python-expr is used to workaround https://developer.blender.org/T74304
```
Comment (veprbl) [inline pkgs/applications/misc/blender/default.nix:167]: This comment is outdated, can be removed.

File: pkgs/applications/misc/blender/default.nix [hunk 2]
```diff
@@ -8,144 +8,168 @@
...
+  ''
+    makeWrapper "${unwrapped}/${blenderExecutable}" "$out/${blenderExecutable}" \
```
Comment (veprbl) [inline pkgs/applications/misc/blender/default.nix:169]:
```suggestion
    makeWrapper "$unwrapped/${blenderExecutable}" "$out/${blenderExecutable}" \
```
This should allow to override the unwrapped blender.

**Review by veprbl — COMMENTED** [review]
I don't know if this is a useful change. I would say that the solution proposed in https://github.com/NixOS/nixpkgs/issues/97250#issuecomment-687728607 is more than appropriate given that we don't allow for plugins themselves to be installed declaratively. I would like to also propose to simply add `pythonPackages.requests` to blender wrapper unconditionally.

**Comment by FRidh** [comment]
>  appropriate given that we don't allow for plugins themselves to be installed declaratively

That is a good remark.

In principle, one could even wrap blender themselves with a Python path like I did here.

**Comment by gebner** [comment]
This should be split into two top-level derivations to allow overriding (and `nix-prefetch-url -A blender-unwrapped.src`).

I think it's a good idea to have the wrapper in nixpkgs.  Writing wrappers for desktop applications is hard to do right, you need to change the desktop files, pick the right python version, make sure that you've got all executables, etc.

**Comment by drupol** [comment]
Hi,

The original issue for which this PR has been created has been fixed already. Therefore, closing this PR.

If you think this should stay open, please let me know.

**Pull request closed without merging**